### ISSUE TITLE
Adapt_grid stuck when running with GPU

### ISSUE THREAD

**Issue by zhangchonglin** [sparta/sparta#216]
## Summary

_When running a test problem with AMR (using `adapt_grid`) and `GPU` (by enabling `KOKKOS`) support, simulation stuck during the first mesh adaptation (command `adapt_grid`). Using the same input script with only `CPU` (different executable), the simulation run fine with several mesh adaptations. Is grid adaptation (through `adapt_grid`) supported with `KOKKOS` GPU?_

## Type of Issue

_Bug Report?_

## SPARTA Version (Bug Report)

_latest sparta version at commit: e0cbf5d_

## Expected Behavior (Bug Report)

The code should run fine when using either GPU or CPU executable.

## Actual Behavior (Bug Report)

Running with 1 GPU and 1 MPI rank with `KOKKOS` enabled executable, the code stuck at mesh adaptation step, as seen below:

```
    2000    4.9428501    59996     1186      744        7     2515 
Loop time of 4.94287 on 1 procs for 2000 steps with 59996 particles

MPI task timing breakdown:
Section |  min time  |  avg time  |  max time  |%varavg| %total
---------------------------------------------------------------
Move    | 0.80875    | 0.80875    | 0.80875    |   0.0 | 16.36
Coll    | 0.41211    | 0.41211    | 0.41211    |   0.0 |  8.34
Sort    | 1.1654     | 1.1654     | 1.1654     |   0.0 | 23.58
Comm    | 1.6928     | 1.6928     | 1.6928     |   0.0 | 34.25
Modify  | 0.60741    | 0.60741    | 0.60741    |   0.0 | 12.29
Output  | 0.25392    | 0.25392    | 0.25392    |   0.0 |  5.14
Other   |            | 0.002454   |            |       |  0.05

Particle moves    = 60815404 (60.8M)
Cells touched     = 62425715 (62.4M)
Particle comms    = 0 (0K)
Boundary collides = 0 (0K)
Boundary exits    = 293 (0.293K)
SurfColl checks   = 1917809 (1.92M)
SurfColl occurs   = 4148 (4.15K)
Surf reactions    = 0 (0K)
Collide attempts  = 1020607 (1.02M)
Collide occurs    = 666506 (0.667M)
Gas reactions     = 0 (0K)
Particles stuck   = 0

Particle-moves/CPUsec/proc: 1.23037e+07
Particle-moves/step: 30407.7
Cell-touches/particle/step: 1.02648
Particle comm iterations/step: 1
Particle fraction communicated: 0
Particle fraction colliding with boundary: 0
Particle fraction exiting boundary: 4.81786e-06
Surface-checks/particle/step: 0.0315349
Surface-collisions/particle/step: 6.82064e-05
Surface-reactions/particle/step: 0
Collision-attempts/particle/step: 0.016782
Collisions/particle/step: 0.0109595
Gas-reactions/particle/step: 0

Particles: 59996 ave 59996 max 59996 min
Histogram: 1 0 0 0 0 0 0 0 0 0
Cells:     13200 ave 13200 max 13200 min
Histogram: 1 0 0 0 0 0 0 0 0 0
GhostCell: 0 ave 0 max 0 min
Histogram: 1 0 0 0 0 0 0 0 0 0
EmptyCell: 0 ave 0 max 0 min
Histogram: 1 0 0 0 0 0 0 0 0 0
Surfs:     261 ave 261 max 261 min
Histogram: 1 0 0 0 0 0 0 0 0 0
GhostSurf: 0 ave 0 max 0 min
Histogram: 1 0 0 0 0 0 0 0 0 0
Adapting grid ...
```

## Steps to Reproduce (Bug Report)

Run with the attached run script:
- running with 6 MPI ranks and only using CPU: simulation case run fine;
- running with 1 GPU and 1 MPI rank: simulation case stuck at mesh adaptation `adapt_grid`.

## Further Information, Files, and Links

The attached files contain a minimum input script and files to reproduce this: [Double_Cone_withAMR_GPU.tar.gz](https://github.com/sparta/sparta/files/5758589/Double_Cone_withAMR_GPU.tar.gz)

**Comment by stanmoore1** [comment]
@zhangchonglin Using `adapt_grid` with Kokkos should work. Thanks for the reproducer, I'll take a look.

**Comment by stanmoore1** [comment]
On the GPU, I see the error:

```
Adapting grid ...
BAD CHILD 0: 2553 2554: 0.0609487 0.0291379 0
ERROR on proc 0: Adapt refine particle could not be mapped to child cell (../grid_adapt.cpp:159)
```

**Comment by stanmoore1** [comment]
@zhangchonglin this issue should be fixed by #217, can you confirm?

**Comment by zhangchonglin** [comment]
Thanks Stan @stanmoore1! I can confirm this issue is fixed with #217.
